### ISSUE TITLE
Remove replication table on upgrade

### ISSUE THREAD

**Issue by ctubbsii** [apache/accumulo#3097]
As a follow up to #3080 , need to add an upgrade step that removes the replication table from ZooKeeper, HDFS, and the metadata table, and removes any metadata from the metadata table pertaining to the replication feature.

**Comment by ctubbsii** [comment]
It may be useful to complete #3098 before completing this, in order to make it easier to add new upgrade code.

**Comment by EdColeman** [comment]
Would this be more appropriate as a manual - post install step? I think using the shell to drop the table would perform most, if not all of the clean-up.  This would put the responsibility that it was "safe" to remove any active replication information on the administrators.

If someone was using replication - they would need to ensure that the replication queue was fully drained before performing the upgrade.  We could abort the upgrade if we can determine that a) replication is active and b) data remains that has not been replicated - this would be similar to they way that we refuse to upgrade if any FATE transactions exist because we do not handle FATE conversions.

**Comment by ctubbsii** [comment]
> Would this be more appropriate as a manual - post install step?

@EdColeman Short answer: No.

Users wouldn't be able to delete the replication table, because they can't delete stuff in the builtin accumulo namespace. Unless we bake in special exceptions for replication, which contrasts with the goal of removing it. Trying to relocate the table to user space is one option, but it is fraught with issues, starting with potential name collisions, the fact that the table ID doesn't match the same base-36 number format as other user tables, and also that you don't know which configuration that might be set on the built-in namespace that needs to be copied over to the new table or namespace on relocation.

But also, the information in this table is very heavily contextualized to the current state of Accumulo and its write ahead logs and garbage collection, which have already been stripped of any relevant code that would be affected by this metadata. So any information in this table isn't going to be up-to-date with the state of Accumulo's WALs anyway... not to mention that there's little to no documentation about the structure of this table that an administrator could use to make sense of its contents.

> If someone was using replication - they would need to ensure that the replication queue was fully drained before performing the upgrade. We could abort the upgrade if we can determine that a) replication is active and b) data remains that has not been replicated - this would be similar to they way that we refuse to upgrade if any FATE transactions exist because we do not handle FATE conversions.

It would be enough to just check that the replication table is marked offline in ZK before proceeding. If it's marked as online in ZK, then we could abort. I don't think it's worth doing anything beyond that to investigate the state of replication during the upgrade process.

**Comment by EdColeman** [comment]
This may need to be done first - or other provision need to be made.  I was testing an incremental update where I removed the ZooKeeper property conversions. I started a 2.1 instance, created and populated a test table and on 3.0 (modified) restart, the following is occurring in the tserver.

```
2022-12-01T16:00:21,353 [scan.LookupTask] WARN : exception while doing multi-scan 
java.lang.RuntimeException: java.lang.ClassNotFoundException: org.apache.accumulo.server.replication.StatusCombiner
```

<details>
  <summary>full trace</summary>

```
java.lang.RuntimeException: java.lang.ClassNotFoundException: org.apache.accumulo.server.replication.StatusCombiner
        at org.apache.accumulo.core.iteratorsImpl.IteratorConfigUtil.loadIterators(IteratorConfigUtil.java:231) ~[accumulo-core-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.apache.accumulo.tserver.tablet.ScanDataSource.createIterator(ScanDataSource.java:229) ~[accumulo-tserver-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.apache.accumulo.tserver.tablet.ScanDataSource.iterator(ScanDataSource.java:120) ~[accumulo-tserver-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.apache.accumulo.core.iteratorsImpl.system.SourceSwitchingIterator.seek(SourceSwitchingIterator.java:231) ~[accumulo-core-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.apache.accumulo.tserver.tablet.TabletBase.lookup(TabletBase.java:358) ~[accumulo-tserver-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.apache.accumulo.tserver.tablet.TabletBase.lookup(TabletBase.java:208) ~[accumulo-tserver-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.apache.accumulo.tserver.scan.LookupTask.run(LookupTask.java:129) ~[accumulo-tserver-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.apache.accumulo.tserver.session.ScanSession$ScanMeasurer.run(ScanSession.java:62) ~[accumulo-tserver-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) ~[accumulo-core-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) ~[accumulo-core-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at java.lang.Thread.run(Thread.java:834) ~[?:?]
Caused by: java.lang.ClassNotFoundException: org.apache.accumulo.server.replication.StatusCombiner
        at java.net.URLClassLoader.findClass(URLClassLoader.java:471) ~[?:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:588) ~[?:?]
        at org.apache.accumulo.start.classloader.AccumuloClassLoader$1.loadClass(AccumuloClassLoader.java:217) ~[accumulo-start-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
        at org.apache.accumulo.core.classloader.ClassLoaderUtil.loadClass(ClassLoaderUtil.java:85) ~[accumulo-core-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.apache.accumulo.core.iteratorsImpl.IteratorConfigUtil.loadClass(IteratorConfigUtil.java:240) ~[accumulo-core-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.apache.accumulo.core.iteratorsImpl.IteratorConfigUtil.loadIterators(IteratorConfigUtil.java:215) ~[accumulo-core-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        ... 12 more

```
</details>

**Comment by EdColeman** [comment]
That's the path I am now heading.  For reference, these are the iterator set for replication in 2.1 (with replication off, never activated.

```
uno ashell -e 'config -np -t accumulo.metadata' | grep repli
default    | table.file.replication ........................ | 0
table      | table.iterator.majc.replcombiner .............. | 9,org.apache.accumulo.server.replication.StatusCombiner
table      | table.iterator.minc.replcombiner .............. | 9,org.apache.accumulo.server.replication.StatusCombiner
table      | table.iterator.scan.replcombiner .............. | 9,org.apache.accumulo.server.replication.StatusCombiner
default    | table.replication ............................. | false

```

**Comment by EdColeman** [comment]
Closed with #3137
